WordPress/src/main/java/org/wordpress/android/ui/prefs/AccountSettingsFragment.java

```diff
@@ -275,29 +277,37 @@ public void onAccountChanged(OnAccountChanged event) {
             return;
         }
 
-        if (event.isError()) {
+        // When account change is caused by password change, progress dialog will be shown (i.e. not null).
+        if (mChangePasswordProgressDialog != null) {
             showChangePasswordProgressDialog(false);
 
-            switch (event.error.type) {
-                case SETTINGS_FETCH_ERROR:
-                    ToastUtils
-                            .showToast(getActivity(), R.string.error_fetch_account_settings, ToastUtils.Duration.LONG);
-                    break;
-                case SETTINGS_POST_ERROR:
-                    ToastUtils.showToast(getActivity(), R.string.error_post_account_settings, ToastUtils.Duration.LONG);
-                    // we optimistically show the email change snackbar, if that request fails, we should
-                    // remove the snackbar
-                    checkIfEmailChangeIsPending();
-                    break;
-            }
-        } else {
-            // When account change is caused by password change, progress dialog will be shown (i.e. not null).
-            if (mChangePasswordProgressDialog != null) {
-                showChangePasswordProgressDialog(false);
+            if (event.isError()) {
+                ToastUtils.showToast(getActivity(), event.error.message, ToastUtils.Duration.LONG);
+                AppLog.e(T.SETTINGS, event.error.message);
+            } else {
                 ToastUtils.showToast(getActivity(), R.string.change_password_confirmation, ToastUtils.Duration.LONG);
+                refreshAccountDetails();
             }
+        } else {
+            showChangePasswordProgressDialog(false);
```

Comment:
Do we need to hide password change progress dialog here? If I'm understanding this correctly, `mChangePasswordProgressDialog` will always be null in this case.

Reply:
Yep, when I added the `if (mChangePasswordProgressDialog != null)` check in be72326, it made the `showChangePasswordProgressDialog(false);` statement in the `else` branch unnecessary.  I remove that statement in e5de395.